pkg/reconciler/namespace/controller.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespace
+
+import (
+	"context"
+	kcertinformer "github.com/knative/serving/pkg/client/injection/informers/networking/v1alpha1/certificate"
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/reconciler"
+	"github.com/knative/serving/pkg/reconciler/namespace/config"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	nsinformer "knative.dev/pkg/injection/informers/kubeinformers/corev1/namespace"
+)
+
+const (
+	controllerAgentName = "namespace-controller"
+)
+
+type configStore interface {
+	ToContext(ctx context.Context) context.Context
+	WatchConfigs(w configmap.Watcher)
+}
+
+// NewController initializes the controller and is called by the generated code
+// Registers eventhandlers to enqueue events.
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+
+	nsInformer := nsinformer.Get(ctx)
+	knCertificateInformer := kcertinformer.Get(ctx)
+
+	c := &Reconciler{
+		Base:                reconciler.NewBase(ctx, controllerAgentName, cmw),
+		nsLister:            nsInformer.Lister(),
+		knCertificateLister: knCertificateInformer.Lister(),
+	}
+
+	impl := controller.NewImpl(c, c.Logger, "Namespace")
+
+	c.Logger.Info("Setting up event handlers")
+	nsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
We can probably reconcile objects that only whitelisted (see https://github.com/knative/serving/blob/8593e2ebf44d306e9f48f0719acd55db84441ac3/pkg/reconciler/autoscaling/kpa/controller.go#L87 for example)

Reply:
That's a good idea, I'll do that.

Reply:
Why? You have that exact check in your reconcile loop? Just bring it here and save time, effort and computing cycles :-)